metadata/src/main/java/org/apache/kafka/controller/ClientQuotaControlManager.java

```diff
@@ -272,4 +276,44 @@ private ApiError validateEntity(ClientQuotaEntity entity, Map<String, String> va
 
         return ApiError.NONE;
     }
+
+    class ClientQuotaControlIterator implements Iterator<List<ApiMessageAndVersion>> {
+        private final long epoch;
+        private final Iterator<Entry<ClientQuotaEntity, TimelineHashMap<String, Double>>> iterator;
+
+        ClientQuotaControlIterator(long epoch) {
+            this.epoch = epoch;
+            this.iterator = clientQuotaData.entrySet(epoch).iterator();
```

Comment:
Since we're getting the entry set of a particular epoch, does this mean we'll be getting consistent snapshots?

Reply:
Yes, snapshots are consistent.